slangpy/benchmarks/test_benchmark_interop.py

```diff
@@ -0,0 +1,449 @@
+    ReportFixture,
+)
+
+ADD_FLOATS = """
+float add_floats(float a, float b) {
+    return a + b;
+}
+"""
+
+ADD_COMPUTE = """
+[shader("compute")]
+[numthreads(32, 1, 1)]
+void compute_main(uint3 tid: SV_DispatchThreadID, StructuredBuffer<float> a, StructuredBuffer<float> b, RWStructuredBuffer<float> res) {
+    res[tid.x] = a[tid.x] + b[tid.x];
+}
+"""
+
+ADD_COMPUTE_4X = """
+[shader("compute")]
+[numthreads(32, 1, 1)]
+void compute_main(uint3 tid: SV_DispatchThreadID, StructuredBuffer<float> a, StructuredBuffer<float> b, RWStructuredBuffer<float> res) {
+    uint idx = tid.x*4;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+}
+"""
+
+ADD_COMPUTE_16X = """
+[shader("compute")]
+[numthreads(32, 1, 1)]
+void compute_main(uint3 tid: SV_DispatchThreadID, StructuredBuffer<float> a, StructuredBuffer<float> b, RWStructuredBuffer<float> res) {
+    uint idx = tid.x*16;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+    idx++;
+    res[idx] = a[idx] + b[idx];
+}
+"""
+
+ADD_COMPUTE_16X_LOOP = """
+[shader("compute")]
+[numthreads(32, 1, 1)]
+void compute_main(uint3 tid: SV_DispatchThreadID, StructuredBuffer<float> a, StructuredBuffer<float> b, RWStructuredBuffer<float> res) {
+    uint base = tid.x*16;
+    for(uint i = 0; i < 16; i++) {
+        uint idx = base + i;
+        res[idx] = a[idx] + b[idx];
+    }
+}
+"""
+
+ADD_COMPUTE_16X_UNROLL = """
+[shader("compute")]
+[numthreads(32, 1, 1)]
+void compute_main(uint3 tid: SV_DispatchThreadID, StructuredBuffer<float> a, StructuredBuffer<float> b, RWStructuredBuffer<float> res) {
+    uint base = tid.x*16;
+    [ForceUnroll]
+    for(uint i = 0; i < 16; i++) {
+        uint idx = base + i;
+        res[idx] = a[idx] + b[idx];
+    }
+}
+"""
+
+
+def test_pytorch_tensor_addition_cpu(benchmark_python_function: BenchmarkPythonFunction):
+    device = helpers.get_torch_device(spy.DeviceType.cuda)
+
+    BUFFER_SIZE = 10
+    import torch
+
+    buffer0 = torch.randn((BUFFER_SIZE,), dtype=torch.float32, device="cuda")
+    buffer1 = torch.randn((BUFFER_SIZE,), dtype=torch.float32, device="cuda")
+
+    def tensor_addition(a: torch.Tensor, b: torch.Tensor):
+        res = a + b
+
+    benchmark_python_function(device, tensor_addition, a=buffer0, b=buffer1)
```

Comment:
Wouldn't it be better to not have to pass a device here? It's not used for the actual benchmark so it would be cleaner to not having to pass it.